internal/backend/db/db.go

```diff
@@ -152,7 +152,7 @@ type DB interface {
 	SaveSignal(ctx context.Context, signalID string, workflowID string, connectionID sdktypes.ConnectionID, filter string) (string, error)
 	GetSignal(ctx context.Context, signalID string) (scheme.Signal, error)
 	RemoveSignal(ctx context.Context, signalID string) error
-	ListSignalsWaitingOnConnection(ctx context.Context, connectionID sdktypes.ConnectionID, filter string) ([]scheme.Signal, error)
+	ListSignalsWaitingOnConnection(ctx context.Context, connectionID sdktypes.ConnectionID) ([]scheme.Signal, error)
```

Comment:
just fyi - i think returning raw scheme data is wrong, we should wrap it somehow.

Reply:
I prefer to work with the schema than adding another layer for nothing currently 
especially not another sdk type

Reply:
no need for sdktype here - it is not exposed outside.

Reply:
i still think the scheme should not be exposed - it might contain internal db data and such. can you add a TODO to fix it?

Reply:
added ticket